test/Governance.test.ts

```diff
@@ -0,0 +1,576 @@
+    await governor.connect(alice).castVote(proposalId, 1);
+    await governor.connect(bob).castVote(proposalId, 1);
+    await governor.connect(charlie).castVote(proposalId, 0);
+
+    // Proposal not yet ready for queue
+    await expect(
+      governor.connect(alice)['queue(uint256)'](proposalId),
+    ).to.be.revertedWith('Governor: proposal not successful');
+
+    // Voting period of 7 days
+    await timeTravel(7);
+
+    // Proposal ready for queue
+    await governor.connect(alice)['queue(uint256)'](proposalId);
+
+    // Proposal not yet ready for execution
+    await expect(
+      governor.connect(alice)['execute(uint256)'](proposalId),
+    ).to.be.revertedWith('TimelockController: operation is not ready');
+
+    // Timelock period of 2 days
+    await timeTravel(2);
+
+    // Executing proposal transfers tokens from treasury to target account
+    await expect(
+      governor.connect(alice)['execute(uint256)'](proposalId),
+    ).to.changeTokenBalances(
+      mentoToken,
+      [david, timelock],
+      [amountToTransfer, -amountToTransfer],
+    );
+  });
+
+  it('should cancel a queued proposal', async function () {
+    const amountToTransfer = parseEther('1000000');
+    const target = david.address;
+    const description = 'Transfer 1m tokens to David';
+
+    const calldata = mentoToken.interface.encodeFunctionData('transfer', [
+      target,
+      amountToTransfer,
+    ]);
+
+    // Create a proposal to transfer tokens from the treasury
+    const proposalId = submitProposal(
+      alice,
+      [governanceAddresses.MentoToken],
+      [0n],
+      [calldata],
+      description,
+    );
+
+    // Vote on the proposal using multiple accounts, with the majority voting YES
+    await governor.connect(alice).castVote(proposalId, 1);
+    await governor.connect(bob).castVote(proposalId, 1);
+    await governor.connect(charlie).castVote(proposalId, 0);
+
+    // Voting period of 7 days
+    await timeTravel(7);
+
+    // Proposal ready for queue
+    await governor.connect(alice)['queue(uint256)'](proposalId);
+
+    const timelockId = timelock.hashOperationBatch(
+      [governanceAddresses.MentoToken],
+      [0],
+      [calldata],
+      ZeroHash,
+      keccak256(toUtf8Bytes(description)),
+    );
+
+    // only the canceller can cancel the proposal
+    const cancellerRole = await timelock.CANCELLER_ROLE();
+    await expect(timelock.connect(alice).cancel(timelockId)).to.be.revertedWith(
+      `AccessControl: account ${alice.address.toLowerCase()} is missing role ${cancellerRole}`,
+    );
+
+    const watchdogAddress = await governanceFactory.watchdogMultiSig();
+    const watchdog = await getImpersonatedSigner(watchdogAddress);
+
+    expect(await timelock.isOperationPending(timelockId)).to.be.true;
+    // watchdog can cancel the proposal
+    await timelock.connect(watchdog).cancel(timelockId);
+    expect(await timelock.isOperationPending(timelockId)).to.be.false;
```

Comment:
should we also check that the proposal itself has status cancelled now with `expect(await governor.state(proposalId)).to.eq(2)`?